packages/components/src/Badge/Badge.component.js

```diff
@@ -19,32 +21,87 @@ function Badge({
 	t,
 	style,
 	className,
+	display = 'large',
+	aslink,
+	edit,
+	icon,
+	white,
 }) {
+	const displayClass = display === 'small' ? 'tc-badge-display-small' : 'tc-badge-display-large';
+
 	const containerClasses = classNames(
 		'tc-badge',
 		theme['tc-badge'],
 		selected && ['tc-badge-selected', theme['tc-badge-selected']],
 		disabled && ['tc-badge-disabled', theme['tc-badge-disabled']],
 		!onDelete && ['tc-badge-readonly', theme['tc-badge-readonly']],
+		[displayClass, theme[displayClass]],
+		aslink && ['tc-badge-aslink', theme['tc-badge-aslink']],
+		edit && ['tc-badge-edit', theme['tc-badge-edit']],
 		className,
 	);
 	const badgeClasses = classNames('tc-badge-button', theme['tc-badge-button']);
-	const labelClasses = classNames('tc-badge-label', theme['tc-badge-label']);
+	const badgeWhiteClasses = classNames(
```

Comment:
Could be only badgeClasses if you use the classname api

```javascript
const badgeClasses = theme(
  'tc-badge-button',
  {
    'tc-badge-white': white,
  }
)
```

Reply:
👍